porcupine/plugins/python_tools.py

```diff
@@ -36,13 +36,23 @@ def run_tool(tool: str, code: str, path: Path | None) -> str:
             [str(python), "-m", tool, "-"],
             check=True,
             stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
             cwd=(Path.home() if path is None else path.parent),
             input=code.encode("utf-8"),
         )
         return result.stdout.decode("utf-8")
     except Exception:
         log.exception(f"running {tool} failed")
-        messagebox.showerror(f"Running {tool} failed", traceback.format_exc())
+        if tool == "black":
+            messagebox.showerror(
```

Comment:
The messagebox should probably not go under `except Exception:`. It should instead catch whatever more specific error gets raised when the subprocess exits with an error status. I think it is `subprocess.CalledProcessError`, but I'm not sure. This way it's less likely to show an error message telling about syntax errors when the code doesn't have syntax errors. Also, I don't see any reason to not do this with `isort`.

So instead of "oh no error for `black` and traceback error for `isort`", I think the logic should be "oh no error for `subprocess.CalledProcessError` and traceback error for other exceptions".